ldms/src/core/ldms.h

```diff
@@ -1211,6 +1540,55 @@ extern const char *ldms_set_name_get(ldms_set_t s);
  */
 extern uint32_t ldms_set_card_get(ldms_set_t s);
 
+/**
+ * \brief Return a copy of the digest of the set schema
+ *
+ * This function returns a copy of the metric set's
+ * schema digest. The \c digest is a SHA256 hash of the
+ * set schema's metric names and types. This has can be
+ * used to confirm that two metric sets have the same
+ * schema.
+ *
+ * The ldms_set_digest_cmp() function should be used to
+ * compare two digests.
+ *
+ * \param s The set handle
+ * \return The schema digest
+ */
+#define LDMS_DIGEST_LENGTH SHA256_DIGEST_LENGTH
+struct ldms_digest_s {
+	unsigned char digest[LDMS_DIGEST_LENGTH];
+};
+typedef struct ldms_digest_s *ldms_digest_t;
```

Comment:
@tom95858 I put this into a structure instead of array typedef because when I did the array typedef:
```c
typedef unsigned char ldms_digest_t[LDMS_DIGEST_LENGTH];
```

The `ldms_digest_t ldms_set_digest_get(set)` signature results in an error: function cannot return array types.